### ISSUE TITLE
Test failure: baseservices\\threading\\interlocked\\compareexchange\\CompareExchangeTClass_1\\CompareExchangeTClass_1.cmd

### ISSUE THREAD

**Issue by v-haren** [dotnet/runtime#41928]
failed in job: [runtime-coreclr gcstress-extra 20200906.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=804112&view=ms.vss-test-web.build-test-results-tab&runId=25467842&paneView=debug&resultId=104635)

CoreCLR Windows_NT arm64 Checked heapverify1 @ Windows.10.Arm64v8.Open

Error message
~~~
Assert failure(PID 6672 [0x00001a10], Thread: 1796 [0x0704]): Consistency check failed: AV in clr at this callstack:
------
CORECLR! WKS::gc_heap::bgc_verify_mark_array_cleared + 0x7C (0x00007fffaabee5ac)
CORECLR! WKS::gc_heap::get_segment + 0x16C (0x00007fffaabfa3fc)
CORECLR! WKS::gc_heap::get_segment_for_uoh + 0x24 (0x00007fffaabfa4d4)
CORECLR! WKS::gc_heap::get_uoh_segment + 0x26C (0x00007fffaabfab8c)
CORECLR! WKS::gc_heap::uoh_get_new_seg + 0x60 (0x00007fffaac0e688)
CORECLR! WKS::gc_heap::allocate_uoh + 0x788 (0x00007fffaabe9fe0)
CORECLR! WKS::gc_heap::try_allocate_more_space + 0x2C8 (0x00007fffaac0e138)
CORECLR! WKS::gc_heap::allocate_more_space + 0x34 (0x00007fffaabe91c4)
CORECLR! WKS::gc_heap::allocate_uoh_object + 0x98 (0x00007fffaabea320)
CORECLR! WKS::GCHeap::Alloc + 0xB4 (0x00007fffaabe2574)
CORECLR! Alloc + 0x144 (0x00007fffaaa817dc)
CORECLR! AllocateString + 0xC8 (0x00007fffaaa82468)
CORECLR! FramedAllocateString + 0x94 (0x00007fffaaaab5c4)
CORECLR! AllocateString_MP_FastPortable + 0x160 (0x00007fffaaaaaf80)
SYSTEM.PRIVATE.CORELIB! System.String.Concat(System.String, System.String)$##6000708 + 0xAC (0x00007fffaa1d88dc)
<no module>! <no symbol> + 0x0 (0x00007fff4b1f0de0)
-----
.AV on tid=0x704 (1796), cxr=000000A40F87E6A0, exr=000000A40F87EA30
FAILED: false

CORECLR! CHECK::Trigger + 0x2D0 (0x00007fffaac25880)
CORECLR! CLRVectoredExceptionHandlerPhase3 + 0x278 (0x00007fffaa9622a8)
CORECLR! CLRVectoredExceptionHandlerPhase2 + 0x64 (0x00007fffaa961e3c)
CORECLR! CLRVectoredExceptionHandler + 0x168 (0x00007fffaa961db0)
CORECLR! CLRVectoredExceptionHandlerShim + 0x1C4 (0x00007fffaa962574)
NTDLL! RtlGetSaclSecurityDescriptor + 0x180 (0x00007fffe046ccb0)
NTDLL! LdrControlFlowGuardEnforced + 0x794 (0x00007fffe042cdc4)
NTDLL! KiUserExceptionDispatcher + 0x24 (0x00007fffe0403434)
CORECLR! WKS::gc_heap::bgc_verify_mark_array_cleared + 0x7C (0x00007fffaabee5ac)
CORECLR! WKS::gc_heap::get_segment + 0x16C (0x00007fffaabfa3fc)
    File: F:\workspace\_work\1\s\src\coreclr\src\vm\excep.cpp Line: 7505
    Image: D:\h\w\BA8709B9\p\CoreRun.exe


Return code:      1
Raw output file:      D:\h\w\BA8709B9\w\A6B6091F\e\baseservices\threading\Reports\baseservices.threading\interlocked\compareexchange\CompareExchangeTClass_1\CompareExchangeTClass_1.output.txt
Raw output:
BEGIN EXECUTION
 cmd /c ..\CompareExchangeTClass\CompareExchangeTClass.cmd
BEGIN EXECUTION
 "D:\h\w\BA8709B9\p\corerun.exe" CompareExchangeTClass.dll "hello world" 
Creating threads
Joining threads
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Expected: 0
Actual: 1
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=D:\h\w\BA8709B9\p
> D:\h\w\BA8709B9\w\A6B6091F\e\baseservices\threading\interlocked\compareexchange\CompareExchangeTClass_1\CompareExchangeTClass_1.cmd
Expected: True
Actual:   False


Stack trace
   at baseservices_threading._interlocked_compareexchange_CompareExchangeTClass_1_CompareExchangeTClass_1_._interlocked_compareexchange_CompareExchangeTClass_1_CompareExchangeTClass_1_cmd() in F:\workspace\_work\1\s\artifacts\tests\coreclr\Windows_NT.arm64.Checked\TestWrappers\baseservices.threading\baseservices.threading.XUnitWrapper.cs:line 19637
~~~

**Comment by VincentBu** [comment]
Failed again in run : [runtime-coreclr gcstress-extra 20210131.1](https://dev.azure.com/dnceng/public/_build/results?buildId=973314&view=ms.vss-test-web.build-test-results-tab&runId=30564130&resultId=106240&paneView=debug)

Failed test : `baseservices\\threading\\interlocked\\compareexchange\\CompareExchangeTClass_1\\CompareExchangeTClass_1.cmd`

CoreCLR windows arm64 Checked heapverify1 @ Windows.10.Arm64v8.Open

**Error message:**
```
Assert failure(PID 220 [0x000000dc], Thread: 10148 [0x27a4]): Consistency check failed: AV in clr at this callstack:
------
CORECLR! WKS::gc_heap::bgc_verify_mark_array_cleared + 0x7C (0x00007ffe43f8381c)
CORECLR! WKS::gc_heap::get_segment + 0x16C (0x00007ffe43f8fb34)
CORECLR! WKS::gc_heap::get_segment_for_uoh + 0x24 (0x00007ffe43f8fc0c)
CORECLR! WKS::gc_heap::get_uoh_segment + 0x26C (0x00007ffe43f90304)
CORECLR! WKS::gc_heap::uoh_get_new_seg + 0x60 (0x00007ffe43fa4158)
CORECLR! WKS::gc_heap::allocate_uoh + 0x788 (0x00007ffe43f7f1a8)
CORECLR! WKS::gc_heap::try_allocate_more_space + 0x2C8 (0x00007ffe43fa3ab8)
CORECLR! WKS::gc_heap::allocate_more_space + 0x34 (0x00007ffe43f7e38c)
CORECLR! WKS::gc_heap::allocate_uoh_object + 0x98 (0x00007ffe43f7f4e8)
CORECLR! WKS::GCHeap::Alloc + 0xB4 (0x00007ffe43f77534)
CORECLR! Alloc + 0x144 (0x00007ffe43e1afe4)
CORECLR! AllocateString + 0xC8 (0x00007ffe43e1bc70)
CORECLR! FramedAllocateString + 0x94 (0x00007ffe43e42504)
CORECLR! AllocateString_MP_FastPortable + 0x160 (0x00007ffe43e41ed0)
SYSTEM.PRIVATE.CORELIB! System.String System.String::Concat(System.String, System.String)$##600072F + 0xA0 (0x00007ffe43343f00)
<no module>! <no symbol> + 0x0 (0x00007ffde458c100)
-----
.AV on tid=0x27a4 (10148), cxr=0000004848DFE810, exr=0000004848DFEBA0
FAILED: false

CORECLR! CHECK::Trigger + 0x2D0 (0x00007ffe43fbb2e0)
CORECLR! CLRVectoredExceptionHandlerPhase3 + 0x274 (0x00007ffe43d00534)
CORECLR! CLRVectoredExceptionHandlerPhase2 + 0x64 (0x00007ffe43d000cc)
CORECLR! CLRVectoredExceptionHandler + 0x168 (0x00007ffe43d00040)
CORECLR! CLRVectoredExceptionHandlerShim + 0x1C4 (0x00007ffe43d00804)
NTDLL! RtlGetSaclSecurityDescriptor + 0x180 (0x00007ffe738cccb0)
NTDLL! LdrControlFlowGuardEnforced + 0x794 (0x00007ffe7388cdc4)
NTDLL! KiUserExceptionDispatcher + 0x24 (0x00007ffe73863434)
CORECLR! WKS::gc_heap::bgc_verify_mark_array_cleared + 0x7C (0x00007ffe43f8381c)
CORECLR! WKS::gc_heap::get_segment + 0x16C (0x00007ffe43f8fb34)
 File: F:\workspace\_work\1\s\src\coreclr\vm\excep.cpp Line: 7505
 Image: D:\h\w\A2B40971\p\CoreRun.exe


Return code: 1
Raw output file: D:\h\w\A2B40971\w\AC7A097D\e\baseservices\threading\Reports\baseservices.threading\interlocked\compareexchange\CompareExchangeTClass_1\CompareExchangeTClass_1.output.txt
Raw output:
BEGIN EXECUTION
 cmd /c ..\CompareExchangeTClass\CompareExchangeTClass.cmd
BEGIN EXECUTION
 "D:\h\w\A2B40971\p\corerun.exe" CompareExchangeTClass.dll "hello world" 
Creating threads
Joining threads
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Expected: 0
Actual: 1
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=D:\h\w\A2B40971\p
> D:\h\w\A2B40971\w\AC7A097D\e\baseservices\threading\interlocked\compareexchange\CompareExchangeTClass_1\CompareExchangeTClass_1.cmd
Expected: True
Actual: False

Stack trace
   at baseservices_threading._interlocked_compareexchange_CompareExchangeTClass_1_CompareExchangeTClass_1_._interlocked_compareexchange_CompareExchangeTClass_1_CompareExchangeTClass_1_cmd()

```

**Comment by PeterSolMS** [comment]
Should be fixed by PR #48295
